copycat/api/src/main/java/org/apache/kafka/copycat/storage/Converter.java

```diff
@@ -21,28 +21,37 @@
 import org.apache.kafka.copycat.data.Schema;
 import org.apache.kafka.copycat.data.SchemaAndValue;
 
+import java.util.Map;
+
 /**
  * The Converter interface provides support for translating between Copycat's runtime data format
- * and the "native" runtime format used by the serialization layer. This is used to translate
- * two types of data: records and offsets. The (de)serialization is performed by a separate
- * component -- the producer or consumer serializer or deserializer for records or a Copycat
- * serializer or deserializer for offsets.
+ * and byte[]. Internally, this likely includes an intermediate step to the format used by the serialization
+ * layer (e.g. JsonNode, GenericRecord, Message).
  */
 @InterfaceStability.Unstable
-public interface Converter<T> {
+public interface Converter {
+
+    /**
+     * Configure this class.
+     * @param configs configs in key/value pairs
+     * @param isKey whether is for key or value
+     */
+    void configure(Map<String, ?> configs, boolean isKey);
 
     /**
      * Convert a Copycat data object to a native object for serialization.
+     * @param topic the topic associated with the data
      * @param schema the schema for the value
      * @param value the value to convert
      * @return
      */
-    T fromCopycatData(Schema schema, Object value);
+    byte[] fromCopycatData(String topic, Schema schema, Object value);
```

Comment:
This is useful info to have during serialization and is included in the `Serializer` interface as well. Useful, e.g., for schema validation/registration.